File: rust/src/leet_code/1614.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let mut max_depth = 0;
        let mut depth = 0;
        for c in s.chars() {
            match c {
                '(' => {
                    depth += 1;
                    max_depth = max_depth.max(depth);
                }
                ')' => {
                    depth -= 1;
                }
                _ => (),
            }
        }
        max_depth
    }
}

File: python/leet_code/1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max_depth = 0
        depth = 0
        for c in s:
            if c == '(':
                depth += 1
                max_depth = max(max_depth, depth)
            elif c == ')':
                depth -= 1
        return max_depth
